jsonrpc/throttling.go

```diff
@@ -1,70 +1,41 @@
 package jsonrpc
 
 import (
-	"container/heap"
+	"context"
 	"errors"
-	"sync"
 	"time"
+
+	"golang.org/x/sync/semaphore"
 )
 
 var errRequestLimitExceeded = errors.New("request limit exceeded")
 
+// Throttling provides functionality which limits number of concurrent requests
 type Throttling struct {
-	requestsPerSeconds int
-	requests           timeQueue
-	lock               sync.Mutex
+	sem     *semaphore.Weighted
+	timeout time.Duration
 }
 
-func NewThrottling(requestsPerSeconds int) *Throttling {
+// NewThrottling creates new throttling and limits number of concurrent requests to maximumConcurrentRequests
+func NewThrottling(maximumConcurrentRequests uint64, timeout time.Duration) *Throttling {
 	return &Throttling{
-		requestsPerSeconds: requestsPerSeconds,
+		sem:     semaphore.NewWeighted(int64(maximumConcurrentRequests)),
+		timeout: timeout,
 	}
 }
 
-func (t *Throttling) AttemptRequest() error {
-	t.lock.Lock()
-	defer t.lock.Unlock()
-
-	currTime := time.Now().UTC()
+// AttemptRequest returns an error if there is more thab
```

Comment:
Finish this sentence :wink:

Check if this is good enough:

```suggestion
// AttemptRequest returns an error if there is more than maximum concurrent requests are currently being executed
```

Reply:
fixed